tree/tree/src/TBranch.cxx

```diff
@@ -1370,9 +1366,204 @@ Int_t TBranch::GetEntry(Long64_t entry, Int_t getall)
                GetBasket(i);
             }
          }
+         // Getting the next basket might reset the current one and
+         // cause a reset of the first / next basket entries back to -1.
+         fFirstBasketEntry = first;
+         fNextBasketEntry = updatedNext;
+      }
+      if (user_buffer) {
```

Comment:
Can we add a comment explaining why if we have a user_basket, basket is not the 'currentBasket'?